server/stream.go

```diff
@@ -58,6 +58,9 @@ type StreamConfig struct {
 	Mirror       *StreamSource   `json:"mirror,omitempty"`
 	Sources      []*StreamSource `json:"sources,omitempty"`
 
+	// Allow applying a subject transform to incoming messages before doing anything else
+	InputSubjectTransform *InputSubjectTransform `json:"input_subject_transform,omitempty"`
```

Comment:
Shorter json name possible?

Reply:
I thought of just input_transform but then thought maybe we want to have transform on other things than subjects later. And also thought since it's only in stream config / stream info it's not critical to optimize the size and as humans will be editing those JSONs a lot of the time I erred on the side of maybe a bit too verbose figuring it couldn't hurt 🤷‍♂️ but yeah it's like 'input' it's the best I could come up with but not especially attached to it...

Reply:
Given it’s “subjects” when setting the interest subjects for the stream naming this input sometning is confusing so maybe subject_transform or even subjects_transform to be very clear

Reply:
Yeah I put input in there to try and denote that it's not just messages from the 'subjects' that get transformed but also those from sources or the mirror, but really for lack of a better term

Reply:
Trying to convey it's a 'stream pre-processing' transform that happens after the subjects/mirror/sources step (which itself can already have another transform, at least for sources)

Reply:
We cant possibly convey all this by choice of a single word :) So seems we should at least just pick a term and stick with it, even better if its one we already use.  It's subject(s).

Reply:
Maybe trying to cram too much meaning in a field name (picking good names can be hard 😅) and "subject_transform" is enough... @derekcollison let me know what name you like best and I'll adjust everywhere (companion branches) for it.

Reply:
@ripienaar fair enough, I will go with "subject_transform" then (unless someone else comes with a better one by tomorrow morning)